DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/DurableFunctionsAnalyzer/SyntaxNodeUtils.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Linq;
+
+namespace DurableFunctionsAnalyzer
+{
+    public static class SyntaxNodeUtils
+    {
+        public static DurableVersion? version;
+
+        public static DurableVersion GetDurableVersion(SemanticModel semanticModel)
+        {
+            if (version != null)
+            {
+                return (DurableVersion)version;
+            }
+
+            var versionTwoInterface = semanticModel.Compilation.GetTypeByMetadataName("Microsoft.Azure.WebJobs.IDurableOrchestrationContext");
+            version = versionTwoInterface != null ? DurableVersion.V2 : DurableVersion.V1;
+            return (DurableVersion)version;
+        }
+        
+        public static bool IsInsideOrchestrator(SyntaxNode node)
+        {
+            if (TryGetMethodDeclaration(out SyntaxNode methodDeclaration, node))
+            {
+                var parameterList = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.ParameterList)).First();
+
+                foreach (SyntaxNode parameter in parameterList.ChildNodes())
+                {
+                    var attributeList = parameter.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
+                    if (attributeList.Count() >= 1 && attributeList.First().ChildNodes().First().ToString().Equals("OrchestrationTrigger"))
```

Comment:
I don't know if we can guarantee that the first attribute is the one that has to have the OrchestratorTrigger. I would imagine most of the time it will be, and if it is not the first, then this will return false negatives over false positives, which is preferable anyways, so i wouldn't highly prioritize that fix.

Reply:
Good catch! I switched this to a for each loop on each AttributeList returned.